mlfinlab/microstructural_features/encoding.py

```diff
@@ -0,0 +1,91 @@
+"""
+Various functions for message encoding (quantile)
+"""
+import numpy as np
+
+
+def encode_tick_rule_array(tick_rule_array: list) -> str:
+    """
+    Encode array of tick signs (-1, 1, 0)
+
+    :param tick_rule_array: (list) of tick rules
+    :return: (str) encoded message
+    """
+    message = ''
+    for element in tick_rule_array:
+        if element == 1:
+            message += 'a'
+        elif element == -1:
+            message += 'b'
+        elif element == 0:
+            message += 'c'
+        else:
+            raise ValueError('Unknown value for tick rule: {}'.format(element))
+    return message
+
+
+def _get_ascii_table() -> list:
+    """
+    Get all ASCII symbols
+
+    :return: (list) of ASCII symbols
+    """
+    # ASCII table consists of 256 characters
+    table = []
+    for i in range(256):
+        table.append(chr(i))
+    return table
+
+
+def quantile_mapping(array: list, num_letters: int = 26) -> dict:
+    """
+    Generate dictionary of quantile-letters based on values from array and dictionary length (num_letters).
+
+    :param array: (list) of values to split on quantiles
+    :param num_letters: (int) number of letters(quantiles) to encode
+    :return: (dict) of quantile-symbol
+    """
+    encoding_dict = {}
+    ascii_table = _get_ascii_table()
+    alphabet = ascii_table[:num_letters]
+    for quant, letter in zip(np.linspace(0.01, 1, len(alphabet)), alphabet):
+        encoding_dict[np.quantile(array, quant)] = letter
+    return encoding_dict
+
+
+def _find_nearest(array: list, value: float) -> float:
+    """
+    Find the nearest element from array to value.
+
+    :param array: (list) of values
```

Comment:
For the doc sake, please, state here that array should be sorted, or add quick is_sorted check (e.g. via https://stackoverflow.com/questions/47004506/check-if-a-numpy-array-is-sorted/47004533)